### PR TITLE
feat: add retry

### LINKED ISSUE #71: Add support for retries
This issue comes from this issue  https://github.com/actions/create-github-app-token/issues/42#issuecomment-1783965170
----
As a rule of thumb, in all code that works with the network, the network will always one day fail. 
For that reason, it will be great to implement a retry logic inside of the action so that the requests made to the GitHub API have a retry logic. 

Also, we can expose the retry parameters to the user.
```
  attempt_limit:
    description: Number of attempts
    required: false
    default: 2
  attempt_delay:
    description: A delay between attempts in ms
    required: false
    default: 0
```

### PR DESCRIPTION
**Pull request by Moser-ss**
resolves #71

- Add p-retry library
- Extract logic to new functions to improve the usage of retry logic

### CODE REVIEW

**Review by gr2m — COMMENTED** [review]
could you please add tests to prevent avoid future regressions? You can throw 500 error responses to test the retries

**Comment by Moser-ss** [comment]
@gr2m I Just added some tests, but I have problems resetting or clearing the mocks we have in the [test/main.js](https://github.com/actions/create-github-app-token/blob/main/tests/main.js#L56-L66)
When I started the test, we already had an intercept defined for the endpoint `GET /repos/{owner}/{repo}/installation"` 
I am not sure if I need or have permission to rework the tests to adjust for the new use case

**Comment by Moser-ss** [comment]
I was able to understand my mistake, I already run the tests locally and I got 100% coverage

File: tests/main-token-get-owner-set-to-user-fail-response.test.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  // Mock installation id request
+  const mockInstallationId = "123456";
+  mockPool
+    .intercept({
+      path: `/orgs/${process.env.INPUT_OWNER}/installation`,
+      method: "GET",
+      headers: {
+        accept: "application/vnd.github.v3+json",
+        "user-agent": "actions/create-github-app-token",
+        // Intentionally omitting the `authorization` header, since JWT creation is not idempotent.
```
Comment (gr2m) [inline tests/main-token-get-owner-set-to-user-fail-response.test.js:17]: alternatively we could mock the date with something like https://www.npmjs.com/package/mockdate, but I don't think it's worth it in this case, the comment by itself is great

**Review by gr2m — APPROVED** [review]
This is great, thank you! I'll add a follow up issue for how we can improve the retry logic in future

**Pull request merged**